app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class CommentsController < ApplicationController
+  def create
+    @comment = @commentable.comments.build(comment_params)
+    @comment.user = current_user
+    @comment.save
+    redirect_to [@commentable], notice: t('controllers.common.notice_create', name: Comment.model_name.human)
+  end
+
+  def destroy
+    @comment = @commentable.comments.find(params[:id])
```

Comment:
```suggestion
    @comment = current_user.comments...
```

のような形式でfindすれば、他人のコメントを削除しようとしたときに例外を発生させて処理を中止することができます

Reply:
レビューありがとうございます！
先に`@comment`取得時に`current_user`に限定してしまうことで例外処理になる、ということですね。
こちらを修正し、合わせて以後のif処理を削除できました。